src/Network/Connection.php

```diff
@@ -351,14 +368,47 @@ public function writeFrame(Frame $stompFrame)
         if (!$this->isConnected()) {
             throw new ConnectionException('Not connected to any server.', $this->activeHost);
         }
-        $data = $stompFrame->__toString();
-        if (@fwrite($this->connection, $data, strlen($data)) !== strlen($data)) {
-            throw new ConnectionException('Was not possible to write frame!', $this->activeHost);
-        }
+        $this->writeData($stompFrame->__toString(), $this->writeTimeout);
         $this->observers->sentFrame($stompFrame);
         return true;
     }
 
+    /**
+     * Write passed data to the stream, respecting passed timeout.
+     *
+     * @param string $data
+     * @param int    $timeout in seconds
+     * @throws ConnectionException
+     */
+    private function writeData($data, $timeout)
+    {
+        $offset = 0;
+        $size = strlen($data);
+        $lastByteTime = microtime(true);
```

Comment:
shouldn't this be inside the loop?

Reply:
I would like to avoid that it's not timing out in case we don't have a successful write at all. So I set the initial value as first timeout challenge.